Helper/Order.php

```diff
@@ -822,13 +824,21 @@ public function dispatchPostCheckoutEvents($order, $quote) {
         $this->_eventManager->dispatch(
             'checkout_submit_all_after', [
                 'order' => $order,
-                'quote' => $quote
+                'quote' => $immutableQuote
             ]
         );
 
+        // Because we used the true session quote for order save, the order's quote ID
+        // does not match the immutable quote that we will keep for historical purposes.
+        // Here we reassign the quote ID to the ID of the same quote copy that we have
+        // passed to the post checkout events.  We do this after the events have processed
+        // to respect any third party code that is perhaps hard-coded to expect
+        // the session quote value in this field.
+        $order->setQuoteId($immutableQuote->getId())->save();
```

Comment:
Since the parent quote is passed into $immutableQuote attribute, this does not do what it was intended to.